pkgs/os-specific/linux/sgx-sdk/default.nix

```diff
@@ -19,59 +18,63 @@
 , perl
 , python3
 , texinfo
-, which
+, validatePkgConfig
 , writeShellScript
+, writeText
 }:
+with lib;
 
 stdenv.mkDerivation rec {
   pname = "sgx-sdk";
-  version = "2.14";
+  version = "2.14.100.2";
 
   src = fetchFromGitHub {
     owner = "intel";
     repo = "linux-sgx";
-    rev = "0cea078f17a24fb807e706409972d77f7a958db9";
+    rev = "sgx_${concatStringsSep "." (take 2 (splitVersion version))}";
```

Comment:
Thanks for being so considerate about reproducibility! You can, however, rest assured—nothing can go wrong here with Nix: the `fetchFromGitHub` call results in a fixed output derivation which has to match the given `sha256`. If someone updates the tag to point to a different commit, two things can happen:

1. Fetching the source fails with a hash mismatch.
2. If the given hash points to a derivation which is already in the store of the executing machine (or in any configured binary cache), it will silently use the original version before the tag was updated in the upstream repository. This easily leads to confusion when updating a `rev` without updating the `sha256` (or at least set it to a non-existing path, e.g., `lib.fakeSha256`).

Overall, what really matters is the hash. If the hash corresponds to the version you want to use, nothing can go wrong.

Still, I cannot confirm that the tag `sgx_2.14` points to a different commit. It's exactly the same:

```shell
# git rev-list -n 1 sgx_2.14
0cea078f17a24fb807e706409972d77f7a958db9
```

https://github.com/NixOS/nixpkgs/blob/a0ab6d84ff85a81feb1081fb46acfdc644feb7d6/pkgs/os-specific/linux/sgx-sdk/default.nix#L33

Reply:
:smile: reproducible builds was the main concern that drove me to `nix` :smile:

Ok, I see, sorry, I overlooked the hash! Totally fine then. As for the commit, I got misled by looking up the tag on GitHub at https://github.com/intel/linux-sgx/commits/sgx_2.14. When checking the tag, you're right, it points to `0cea078f17a24fb807e706409972d77f7a958db9`.